### PR TITLE
Delete split jsonl files and save hdf5 directly to output_path

### PR DESCRIPTION
**Pull request by snova-zoltanc**
## Summary
Adding a default to delete the split jsonl files so they do not take up extra storage space. Give a flag option to not delete the hdf5. Since this is the new default we are also updating the code to place the hdf5 files directly in the output directory instead of creating a separate sub-directories under the output directory called hdf5. 

## PR Checklist
- [ X] My PR is less than 500 lines of code
- [ X] I have added sufficient comment as docstrings in my code
- [ X] I have made corresponding changes to the documentation
- [X ] I have written unit-tests to test all of my code

### CODE REVIEW

File: generative_data_prep/__main__.py [hunk 1]
```diff
@@ -132,6 +132,13 @@ def add_pipeline_args(subparser: argparse.ArgumentParser):
         help="The number of CPU workers to multi-process run tokenization over, if the previous run failed you need to \
         decrease this number.",  # noqa: E251
     )
+    subparser.add_argument(
+        "--do_not_delete_split_jsonls",
```
Comment (snova-virens) [inline generative_data_prep/__main__.py:136]: super nit: Wanna call this variable `keep_split_jsonls`. Slightly easier to read.

**Review by snova-virens — APPROVED** [review]
Looks Great!  Thanks Zoli!

**Review by snova-virens — APPROVED** [review]
Reapproved!

**Comment by snova-zoltanc** [comment]
Hi @pcompieta thanks so much for bringing this to our attention, sorry for not updating the docs and that the dataset upload failed. 

We will fix this immediately.

**Pull request merged**